packages/api-explorer/src/form-components/MultiSchemaField.jsx

```diff
@@ -0,0 +1,319 @@
+  for (let i = 0; i < options.length; i++) {
+    const option = options[i];
+    if (option.properties?.[property]) {
+      return option.properties[property];
+    }
+
+    if (option.allOf?.[0]?.properties?.[property]) {
+      return option.allOf[0].properties[property];
+    }
+  }
+
+  return null;
+}
+
+/**
+ * rjsf/oas-form adds an empty format field to the schema if there isn't one. We don't want that, to ensure we can
+ * identify equality with the schema component
+ */
+function cleanDiscriminatorSchema(schema) {
+  const copy = { ...schema };
+  if (copy.format === '') {
+    delete copy.format;
+  }
+  return copy;
+}
+
+/**
+ * This is a fun one. So here we go.
+ *
+ * If we want to generate a list of options for the discriminator drop down, we look at the schemas in the oneOf/anyOf.
+ * Unfortunately the "value" of the discriminator is the name of that schema as referenced in the components/schemas
+ * list. This is unfortunate because by the time we recieve the schema in this code it has already been dereferenced and
+ * we've lost the json pointer.
+ *
+ * The solution is to compare the schema objects in the oneOf/anyOf to everything in the components/schemas list and use
+ * that to identify the names we should use.
+ *
+ * The final order of this is SUPER IMPORTANT, because the existing multischema component renders the options in the
+ * numeric order of the schemas in the oneOf/anyOf. So we need the values to appear in that same order, otherwise you
+ * might have a name render the wrong schema.
+ *
+ * If we wanted to drop the order requirement, we would need to investigate further the mix between the incoming
+ * options, and the list of schemas we generate. Maybe we could store a key value pair of name=>schema, of valid
+ * options, and rely exclusively on that for this entire file. It would probably clean this code up a little bit.
+ */
+function getSchemaOptions(discriminatorSchemas, componentSchemas) {
+  const options = [];
+
+  Object.keys(discriminatorSchemas).forEach(dsKey => {
+    const discSchema = discriminatorSchemas[dsKey];
+    Object.keys(componentSchemas).forEach(componentKey => {
+      if (deepEquals(cleanDiscriminatorSchema(discSchema), componentSchemas[componentKey])) {
+        options.push({
+          label: componentKey,
+          value: componentKey,
+        });
+      }
+    });
+  });
+
+  return options;
+}
+
+/**
+ * See getSchemaOptions for more thorough details. This basicaly does that, but uses the aliases in the mapping as
+ * values instead of the schema keys
+ */
+function getMappingOptions(mapping, discriminatorSchemas, componentSchemas) {
+  const validSchemas = {};
+
+  // Get a list of valid schemas listed in the mapping and format it to look like its own component schema object, but
+  // with the mapping keys instead of real keys so we can reuse getSchemaOptions.
+  Object.keys(mapping).forEach(key => {
+    const match = mapping[key].match(/#\/components\/schemas\/([^/]+)/);
```

Comment:
Do we need to worry about `#/components/parameters`, `#/components/requestBodies`, or any other of the [other supported ones](https://swagger.io/docs/specification/components/)?

Reply:
Discriminator can only map to schema objects, so we shouldn't have to worry about anything else.